### PR TITLE
net: ipv4: Add support for conflict detection (RFC 5227)

### PR DESCRIPTION
**Pull request by rlubos**
This PR implements (optional) IPv4 conflict detection, as specified in [RFC 5227](https://datatracker.ietf.org/doc/html/rfc5227).

The implementation is based on the solution we had in IPv4 autoconf module, implementing [RFC 3927](https://datatracker.ietf.org/doc/html/rfc3927), which was just a more specialized case of a conflict detection described in RFC 5227.

IPv4 autoconf and DHCP has been integrated with the conflict detection events, so that new address can be generated/requested when a conflict is reported.

### CODE REVIEW

File: subsys/net/lib/shell/events.c [hunk 1]
```diff
@@ -256,6 +256,18 @@ static char *get_l3_desc(struct event_msg *msg,
...
+		*desc2 = "ok";
+		info = net_addr_ntop(AF_INET, msg->data, extra_info,
+				     extra_info_len);
+		break;
+	case NET_EVENT_IPV4_ACD_FAILED:
+		*desc = "IPv4 ACD";
+		*desc2 = "fail";
+		info = net_addr_ntop(AF_INET, msg->data, extra_info,
+				     extra_info_len);
+		break;
```
Comment (jukkar) [inline subsys/net/lib/shell/events.c:270]: Small nit. We should/could add information about the ACD enable/disable status to `subsys/net/lib/shell/ipv4.c` which prints information about IPv4 configuration.

**Review by jukkar — COMMENTED** [review]
LGTM, minor fix needed for CI to pass.

File: subsys/net/lib/shell/events.c (hunk 1 above)
Comment (rlubos) [reply subsys/net/lib/shell/events.c:270]: Good point, added.

**Comment by rlubos** [comment]
Just a rebase, hopefully CI issues should be resolved now.

**Review by jukkar — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by maass-hamburg — APPROVED** [review]

**Pull request merged**